### DISCUSSION TITLE
Union type for SelectItem[value] property?

### DISCUSSION THREAD

**Discussion by dshmatok** [mantinedev/mantine#1551] [Q&A] [answered]
Hi! First of all, thanks for the amazing Mantine library! :)
I want to aks: why was the `string` type chosen for the property `SelectItem[value]`?

https://github.com/mantinedev/mantine/blob/37adbfc7c16d088e1fabef3947d385d145ddcff0/src/mantine-core/src/components/Select/types.ts#L6-L12

If I have an array of `number` values, then I have to perform an explicit conversion of the value type to a `string` and then invert it back to a `number` if I need to continue working with the `number` type.

Has the possibility of combining the `number | string` types been considered?
Thanks!

**Comment by rtivital** [comment, marked as answer]
Supporting additional types will make it significantly harder to maintain the component. It also keeps the component consistent with the rest of the components that accept `data` prop with the same format (Autocomplete, SegmentedControl, etc.). Converting types on your side should not be a huge job, so I won't add this feature to the library to save time for other features.
